gene2phenotype_project/gene2phenotype_app/views/search.py

```diff
@@ -134,47 +141,86 @@ def get_queryset(self):
-            for lgd in queryset:
+            if search_type != 'draft':
+                for lgd in queryset:
                 # If the user is not logged in, only show visible panels and remove disputed and refuted records
-                if user.is_authenticated == False:
-                    lgdpanel_select = LGDPanel.objects.filter(
-                        Q(lgd=lgd, panel__is_visible=1, is_deleted=0)
-                        & ~Q(lgd__confidence__value='disputed') & ~Q(lgd__confidence__value='refuted'))
-                else:
-                    lgdpanel_select = LGDPanel.objects.filter(lgd=lgd, is_deleted=0)
-
-                lgd_panels = []
-                for lp in lgdpanel_select:
-                    lgd_panels.append(lp.panel.name)
-
-                # Add new property to LGD object
-                lgd.panels = lgd_panels
-                if lgd_panels:
-                    new_queryset.append(lgd)
+                    if user.is_authenticated is False:
+                        lgdpanel_select = LGDPanel.objects.filter(
+                            Q(lgd=lgd, panel__is_visible=1, is_deleted=0)
+                            & ~Q(lgd__confidence__value='disputed') & ~Q(lgd__confidence__value='refuted'))
+                    else:
+                        lgdpanel_select = LGDPanel.objects.filter(lgd=lgd, is_deleted=0)
+
+                    lgd_panels = []
+                    for lp in lgdpanel_select:
+                        lgd_panels.append(lp.panel.name)
+
+                    # Add new property to LGD object
+                    lgd.panels = lgd_panels
+                    if lgd_panels:
+                        new_queryset.append(lgd)
+            else:
+                return queryset
 
         return new_queryset
 
     def list(self, request, *args, **kwargs):
         queryset = self.get_queryset()
+        serializer = self.get_serializer_class()
+
+        
         list_output = []
+        if issubclass(serializer, LocusGenotypeDiseaseSerializer):
+            for lgd in queryset:
+                data = { 'id':lgd.stable_id.stable_id,
+                        'gene':lgd.locus.name,
+                        'genotype':lgd.genotype.value,
+                        'disease':lgd.disease.name,
+                        'panel':lgd.panels
+                    }
+                list_output.append(data)
+            paginated_output = self.paginate_queryset(list_output)
+
+            if paginated_output is not None:
+                return self.get_paginated_response(paginated_output)
+
+            return Response({"results": list_output, "count": len(list_output)})
+        else:
+            for c_data in queryset:
+                data = {
+                    "id" : c_data.stable_id.stable_id,
+                    "gene": c_data.gene_symbol,
+                    "date_created": c_data.date_created,
+                    "date_last_updated": c_data.date_last_update,
+                    "curator": c_data.user_name,
```

Comment:
Sometimes, the username is not clear. It would be better to return First and Last name.